service/geopmdpy/varrun.py

```diff
@@ -0,0 +1,519 @@
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+"""Interface for /var/run/geopm-service
+
+Module used to read and write the files in /var/run/geopm-service.
+These files maintain the state used by geopmd to track ongoing client
+sessions.  These files are loaded at geopmd start-up time and enable
+the daemon to cleanly restart.
+
+"""
+
+import os
+import sys
+import stat
+import json
+import jsonschema
+import glob
+import psutil
+import uuid
+
+
+GEOPM_SERVICE_VAR_PATH = '/var/run/geopm-service'
+
+class ActiveSessions(object):
+    """Class that manages the session files for the service
+
+    The state about active sessions opened with geopmd by a client
+    processes is stored in files in /var/run/geopm-service.  These
+    files are loaded when the geopmd process starts.  The files are
+    modified each time a client opens a session, closes a session,
+    requests write permission, or starts a batch server.  The class is
+    responsible for managing file access permissions, and atomic file
+    creation.
+
+    """
+
+    def __init__(self, var_path=GEOPM_SERVICE_VAR_PATH):
+        """Create an ActiveSessions object that tracks geopmd session files
+
+        The geopmd session files are stored in the directory
+
+            "/var/run/geopm-service"
+
+        by default, but the user may specify a different path.  The
+        creation of an ActiveSessions object will make the directory
+        if it does not exist.  This directory is created with
+        restricted access permissions (mode: 0o700).
+
+        If the path points to a symbolic link, the link will be
+        renamed and a warning is printed to the syslog.  The directory
+        will then be created in its place.  The warning message will
+        display the target of the symbolic link. If the path points
+        to a directory which is not accessible, the directory will be
+        renamed and a warning is printed to the syslog. The warning
+        message will display the reason why the directory was not
+        accessible, display the permissions or user and group of the directory being
+        deleteted.
+
+        When an ActiveSessions object is created and the directory
+        already exists the owner of the directory will be set to the
+        geopmd user and the permissions will be set to mode 0o700.
+        All files matching the pattern
```

Comment:
Is this still accurate, or are we deleting and re-creating?

Reply:
We are moving any files that do not meet requirements to uuid4-tmp versions